src/services/refactors/convertArrowFunctionOrFunctionExpression.ts

```diff
@@ -0,0 +1,192 @@
+                Debug.fail("invalid action");
+                break;
+        }
+
+        return undefined;
+    }
+
+    function getFunctionInfo(file: SourceFile, startPosition: number): FunctionInfo | undefined {
+        const token = getTokenAtPosition(file, startPosition);
+        let maybeFunc;
+
+        maybeFunc = getArrowFunctionFromVariableDeclaration(token.parent);
+        if (!!maybeFunc) return { selectedVariableDeclaration: true, func: maybeFunc };
+
+        maybeFunc = getContainingFunction(token);
+        if (!!maybeFunc && (isFunctionExpression(maybeFunc) || isArrowFunction(maybeFunc)) && !rangeContainsRange(maybeFunc.body, token)) {
+            return { selectedVariableDeclaration: false, func: maybeFunc };
+        }
+
+        return undefined;
+    }
+
+    function isSingleVariableDeclaration(parent: Node): parent is VariableDeclarationList {
+        return isVariableDeclaration(parent) || (isVariableDeclarationList(parent) && parent.declarations.length === 1);
+    }
+
+    function getArrowFunctionFromVariableDeclaration(parent: Node): ArrowFunction | undefined {
+        if (!isSingleVariableDeclaration(parent)) return undefined;
+        const variableDeclaration = isVariableDeclaration(parent) ? parent : parent.declarations[0];
+
+        const initializer = variableDeclaration.initializer;
+        if (!initializer || !isArrowFunction(initializer)) return undefined;
+        return initializer;
+    }
+
+    function convertToBlock(body: ConciseBody): Block {
+        if (isExpression(body)) {
+            return createBlock([createReturn(body)], /* multiLine */ true);
+        }
+        else {
+            return body;
+        }
+    }
+
+    function getVariableInfo(func: FunctionExpression | ArrowFunction): VariableInfo | undefined {
+        const variableDeclaration = func.parent;
+        if (!isVariableDeclaration(variableDeclaration) || !isVariableDeclarationInVariableStatement(variableDeclaration)) return undefined;
+
+        const variableDeclarationList = variableDeclaration.parent;
+        const statement = variableDeclarationList.parent;
+        if (!isVariableDeclarationList(variableDeclarationList) || !isVariableStatement(statement) || !isIdentifier(variableDeclaration.name)) return undefined;
+
+        return { variableDeclaration, variableDeclarationList, statement, name: variableDeclaration.name };
+    }
+
+    function getEditInfoForConvertToAnonymousFunction(context: RefactorContext, func: FunctionExpression | ArrowFunction): RefactorEditInfo {
+        const { file } = context;
+        const body = convertToBlock(func.body);
+        const newNode = createFunctionExpression(func.modifiers, func.asteriskToken, /* name */ undefined, func.typeParameters, func.parameters, func.type, body);
+        const edits = textChanges.ChangeTracker.with(context, t => t.replaceNode(file, func, newNode));
+        return { renameFilename: undefined, renameLocation: undefined, edits };

```

Comment:
It may be better for these functions to just return the edits, and some outer function to wrap them in `RefactorEditInfo`.